bob-access/src/extractor.rs

```diff
@@ -40,14 +40,25 @@ impl<T: Extractor> ExtractorExt for T {
 
     fn extract_basic(&self) -> Result<Credentials, Error> {
         let mut builder = prepare_builder(self)?;
-        match (self.get_header("username")?, self.get_header("password")?) {
-            (Some(username), Some(password)) => {
-                let creds = builder
-                    .with_username_password(username, password)
-                    .build();
-                Ok(creds)
-            },
-            (None, None) => {
+        let auth_header = self.get_header("authorization")?;
+        let mut err = "missing authorization header";
+        if let Some(auth_header) = auth_header {
+            let mut parts = auth_header.split_whitespace();
+            if let (Some("Basic"), Some(credentials)) = (parts.next(), parts.next()) {
```

Comment:
I think this should be extracted into separate function, "find_username_password" or something similar